src/common/buffer.cc

```diff
@@ -1613,7 +1613,7 @@ using namespace ceph;
       len -= need;
     }
     if (len) {
-      ptr bp = buffer::create_page_aligned(len);
+      auto& bp = ptr_node::create(buffer::create_page_aligned(len));
       bp.zero(false);
       append(std::move(bp));
```

Comment:
I don't understasnd the lifecycle here.  ptr_node::create allocates on the heap and returns a ref.. but then the ref is passed via std::move to append and push_back and finally passed into a new ptr_node::create() call.  that seems like it would (1) leak the allocation and (2) like it needlessly copies to a new instance of the object, instead of just linking the existing ptr_node we have here into the list. what am i missing?

Reply:
You're right, there is a problem in the intermediary commit. Finally we're switching to the `std::unique_ptr`-taking variant of `bl::push_back`:

```cpp
  void buffer::list::append_zero(unsigned len)
   {
    // ...
    if (len) {
      auto bp = ptr_node::create(buffer::create_page_aligned(len));
      bp->zero(false);
      push_back(std::move(bp));
    }
  }
```

I will fix that in the morning. Thanks for catching!

Reply:
Retriggered the ARM64 build because of: `/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/librbd/managed_lock/test_mock_GetLockerRequest.cc:106:1: internal compiler error: Segmentation fault`